### DISCUSSION TITLE
Why is vitest faster than jest

### DISCUSSION THREAD

**Discussion by spicylemonhaha** [vitest-dev/vitest#1635] [Q&A] [answered]
Why is vitest faster than jest, can anyone give me a direction to study or tell me the reason?

**Comment by bliu13** [comment]
This was what I found in the [docs](https://vitest.dev/guide/why.html#adding-vitest-to-your-project), `Vitest cares a lot about performance and uses Worker threads to run as much as possible in parallel. Some ports have seen test running an order of magnitude faster.`

**Comment by Demivan** [comment]
With proper configuration single run of Vitest will be on par with Jest.

For me single run of Vitest was faster because I stopped using Babel to process Typescript files.

And Vitest is much faster in watch mode as it uses Vite. https://twitter.com/antfu7/status/1468233216939245579

**Reply by spicylemonhaha** [reply]
Can I still enjoy `And Vitest is much faster in watch mode as it uses Vite` privilege if my project is not built by vite?

**Reply by Demivan** [reply]
Yes. Vitest uses Vite internally for code transformation and hot reloading of modules even if you don't use Vite yourself.

**Comment by spicylemonhaha** [comment]
However, as far as I know jest is also multi-threaded. So the reason for the performance difference is that jest does not open as many threads as vitest?

**Comment by nickserv** [comment, marked as answer]
Vite (and therefore Vitest) builds with esbuild which is a native binary compiled with Go. Jest builds with webpack which is a JS package with way more overhead and slower IO. There are other decisions Vite and Vitest make to improvement performance, but this is the main thing separate the new generation of JS tools (Vite, Parcel, esbuild, swc, etc.) from the old generator (webpack, Rollup, etc.) IMO.

**Reply by spicylemonhaha** [reply]
Maybe, It was because of these. Thank you very much.
